### ISSUE TITLE
Auto-release job failing due to conflicting branch names

### ISSUE THREAD

**Issue by dsimansk** [knative-sandbox/kn-plugin-func#801]
This's sort of a bug report wrt CI infra. I'd like to capture my thought on auto-release job failures, that we spotted last week.

Log of the job e.g.:
https://prow.knative.dev/view/gs/knative-prow/logs/ci-knative-sandbox-kn-plugin-func-auto-release/1488124573319696384

IMO, the root cause of the failure is branch conflicting naming to what the `release.sh` expects.  The `v` (in release-v0.20.1) in former bnrach names breaks the calculation of next version to be auto-released, variable calculated [here](https://github.com/knative/hack/blob/main/release.sh#L170). The order is broken because, the `0.21` value should be last ideally, but as 

There are 2 options:
1. Remove the conflicting branches and keep only tags
2. Modify `release.sh` script - I'm not sure if that's feasible, but seems safe since there shouldn't be any "v" the other project depend on :crossed_fingers: 

/cc @lance @rhuss

```console
➜  kn-plugin-func git:(main) { (git branch -r | grep upstream/release-) ; (git branch | grep release-); } | cut -d '-' -f2 | sort -V | uniq         
0.21
v0.6.3
v0.7.0
v0.8.0
v0.9.0
v0.10.0
v0.11.0
v0.11.1
v0.12.0
v0.12.1
v0.12.2
v0.13.0
v0.14.0
v0.15.0
v0.15.1
v0.16.0
v0.16.1
v0.17.0
v0.17.1
v0.18.0
v0.19.0
v0.19.1
v0.20.0
```

E.g. with removing the `v` it looks better.
```console
➜  kn-plugin-func git:(main) { (git branch -r | grep upstream/release-) ; (git branch | grep release-); } | cut -d '-' -f2 | cut -d 'v' -f2 | sort -V | uniq
0.6.3
0.7.0
0.8.0
0.9.0
0.10.0
0.11.0
0.11.1
0.12.0
0.12.1
0.12.2
0.13.0
0.14.0
0.15.0
0.15.1
0.16.0
0.16.1
0.17.0
0.17.1
0.18.0
0.19.0
0.19.1
0.20.0
0.21
```

**Comment by lance** [comment]
I noticed this too last week before I had to leave, and wondered myself what the best way forward would be. Thank you for taking the initiative here! My gut instinct was not to remove the branches, but instead find some way to rename them. However, I think that might introduce some unnecessary work/complexity, and now I am leaning towards just deleting the branches.

Typically, we would want to maintain branches so we can apply fixes on patch releases for previous versions. But all of these previous versions are pre-1.0 and we have essentially no intention of backporting any fixes prior to those in `v0.21.2`. But I'd like to hear what @rhuss and others think about this before doing anything.

My suggestion...

- Check to be sure all existing version tags exist on `main`. They should, but if for some reason the tag got created on a commit down the line in a branch, we should move it to the branch head.
- Delete all branches prior to `release-0.21` but keep tags
- ?
- Profit!

**Comment by dsimansk** [comment]
Yep, I'd go with your suggestion. Until tags are untouched, release branches should be recoverable. 

@rhuss a little heads up, all the branches in question are protected of course. :superhero:

**Comment by lance** [comment]
@rhuss I think that covers it. Thank you!
